costmap_2d/include/costmap_2d/obstacle_layer.h

```diff
@@ -65,6 +65,11 @@ class ObstacleLayer : public CostmapLayer
   ObstacleLayer()
   {
     costmap_ = NULL;  // this is the unsigned char* member of parent class Costmap2D.
+    footprint_clearing_enabled_ = true;
```

Comment:
Why did you initialize variables in the ctr?
Commonly, we write
 ```ObstacleLayer() : footprint_clearing_enabled_(true)``` 
or
 ```bool footprint_clearing_enabled_{true} // C++11```.

Reply:
Modify as indicated.

Reply:
OK.